### ISSUE TITLE
syntax error with LAYOUT_WITH_LEGEND and LAYOUT_AS_SKETCH

### ISSUE THREAD

**Issue by coffeefedora** [plantuml-stdlib/C4-PlantUML#23]
I am using PlantUML extension in Visual Studio Code.

Seems to work fine until i try the Layout with legend or layout as sketch.

This specific test is with the C4_Context.puml file.  I get the error with all files though.

I get a syntax error in the preview window:

...
.. (skipping 85 lines)
...

skinparam database<<external_system>> {
   StereotypeFontColor #FFFFFF
   FontColor #FFFFFF
   BackgroundColor #999999
   BorderColor #8A8A8A
}

Turns out that text is just prior to the macro definition for the LAYOUT_WITH_LEGEND.  So something is not right with the macro, or the plantUML extension maybe has a bug with multi line macros?

**Comment by majj** [comment]
the same issue.

**Comment by sunduowei** [comment]
the same issue. but it worked weeks before

**Comment by sunduowei** [comment]
the same issue

**Comment by sunduowei** [comment]
C4_Context.puml
```
!function layout_with_legend()
hide stereotype 
legend right
|=              |= Type |
|<PERSON_BG_COLOR>      | person |
|<EXTERNAL_PERSON_BG_COLOR>      | external person |
|<SYSTEM_BG_COLOR>   | system |
|<EXTERNAL_SYSTEM_BG_COLOR>      | external system |
endlegend
!endfunction
```
xxx.puml

use layout_with_legend() instead of LAYOUT_WITH_LEGEND

**Comment by adrianvlupu** [comment]
You can call `LAYOUT_WITH_LEGEND()` using `()`, like a function, and it works.

**Comment by coffeefedora** [comment]
ah!  thank you

**Comment by coffeefedora** [comment]
the same approach also fixes LAYOUT_AS_SKETCH fyi

**Comment by Tinkerer-** [comment]
Thanks a lot, had the same issue. Will this be fixed (either in documentation or in code)?

**Comment by tianlinzx** [comment]
still can't resolve the problem, even if I change the micro into function .
any other ideas ?

**Comment by haakoo** [comment]
> still can't resolve the problem, even if I change the micro into function .
> any other ideas ?

You need to add braces when you call the _function_, e.g.:

```plantuml
@startuml LAYOUT_TOP_DOWN Sample
!includeurl https://raw.githubusercontent.com/RicardoNiepel/C4-PlantUML/master/C4_Container.puml

/' Not needed because this is the default '/
LAYOUT_TOP_DOWN
LAYOUT_WITH_LEGEND()

Person(admin, "Administrator")
System_Boundary(c1, 'Sample') {
    Container(web_app, "Web Application", "C#, ASP.NET Core 2.1 MVC", "Allows users to compare multiple Twitter timelines")
}
System(twitter, "Twitter")

Rel(admin, web_app, "Uses", "HTTPS")
Rel(web_app, twitter, "Gets tweets from", "HTTPS")
@enduml
```

**Comment by josemarcosdevops** [comment]
The same issue.

It works when I use like a function.

LAYOUT_WITH_LEGEND()

**Comment by Potherca** [comment]
Resolved in #29 by updating the example in the README.
